src/test/ui/lint/must_use-tuple.rs

```diff
@@ -0,0 +1,5 @@
+#![deny(unused_must_use)]
+
+fn main() {
+    (Ok::<(), ()>(()),); //~ ERROR unused `std::result::Result` that must be used
```

Comment:
What happens when you have a multi-element tuple where multiple of the elements are must-use? Do you get a separate warning for each of them?

Also, it would be good to call out that the warning is happening to a tuple member in the return type so that it's clear where in the return type `Result` appears.

Reply:
You get a single warning if any of the types are `#[must_use]`. It would be good to point out the specific element causing it — I'll try to get around to fixing that tomorrow.